### PR TITLE
Easy way of inserting rows and columns into tables using code

### PR DESCRIPTION
**Pull request by numeredev**
### SUMMARY
Implements necessary changes for #69

*Reviewers:* @numere-org/maintainers

### IMPLEMENTATION
* Implementation: Implemented as regular table methods. Added `TAB().insertcells()`, `TAB().insertrows()`, `TAB().insertcols()` and the same for `TAB().removeXYZ()`. Added `TAB().reorderrows()` and `TAB().reordercols()` for reordering rows and columns.
* Implementation test: Every new table method was tested manually at least once

### DOCUMENTATION
* [x] ChangesLog updated
* [x] Code changes commented
* **Documentation articles:**
    * [x] corresponding documentation articles updated
    * [ ] new documentation articles created
    * [ ] not needed
* **Language files:**
    * [x] corresponding language files updated
    * [ ] not needed

----------------

### TESTS BY REVIEWERS
* [x] Added to the automatic SW tests https://github.com/numere-org/SW-Tests/pull/5
* [ ] Tested manually

### CODE REVIEW

**Comment by numeredev** [comment]
Testing this is quite as simple as using the new table method and comparing the results with the expected behavior, e.g.

```
TAB().insertcols(...);
assert TAB().cols == ...;
assert TAB().size.cols.every(...) == 0;
```

**Comment by adjiap** [comment]
I see a possible bug on .removerows() (or is it actually a feature?)

```
matop mantab_same() = {{1, 2}, {3, 4}};
mantab_same().insertrows(2, 4);
mantab_same().removerows(2, 2);
```
has the same result as 

```
matop mantab_diff() = {{1, 2}, {3, 4}};
mantab_diff().insertrows(2, 4);
mantab_diff().removerows(2, 3);
```
image

Is that what's wanted? I would see the second one (diff) is the correct way to use it, but the first one doesn't say that it's wrong, or that it "just" removes the second row

**Comment by numeredev** [comment]
> I see a possible bug on .removerows() (or is it actually a feature?), and very likely also on removecols()
> 
> 
> 
> ```
> 
> matop mantab_same() = {{1, 2}, {3, 4}};
> 
> mantab_same().insertrows(2, 4);
> 
> mantab_same().removerows(2, 2);
> 
> ```
> 
> has the same result as 
> 
> 
> 
> ```
> 
> matop mantab_diff() = {{1, 2}, {3, 4}};
> 
> mantab_diff().insertrows(2, 4);
> 
> mantab_diff().removerows(2, 3);
> 
> ```
> 
> image
> 
> 
> 
> Is that what's wanted? I would see the second one (diff) is the correct way to use it, but the first one doesn't say that it's wrong, or that it "just" removes the second row

That's not a bug, the removing methods have a different Syntax. In fact, you just removed two rows  each time as it expects a vector of row indices to be delete.

File: changes.tmp [hunk 1]
```diff
@@ -15,6 +15,13 @@ Added	The functions "to_utf8()" and "to_ansi()" can be used to convert to and fr
 Added	Folding has been enabled for LaTeX files.
 Added	The terminal now also proposed global variables for auto completion.
 Applied	The revision of large files won't be created by calculating a DIFF. Instead, the whole file content is saved.
+Added	New table methods for insertion of cells ("TAB().insertXYZ()"), removing of cells ("TAB().removeXYZ()") and removing unnecessary trailing cells and columns ("TAB().shrink") have been added.
```
Comment (adjiap) [inline changes.tmp:18]: Not really needed to be fixed, but a question. 

Is it okay to add the revision in the middle of the changes.tmp? cause it's no longer chronological

**Review by adjiap — APPROVED** [review]
I don't have much to comment on the changes in kernel/core. Most comments are in the Testing PR

File: changes.tmp (hunk 1 above)
Comment (numeredev) [reply changes.tmp:18]: That's totally fine. In the end, the entries will be sorted alphabetically nevertheless.

**Pull request merged**